### PR TITLE
Adds debuffs for hunger and thirst motives, removes dropped clicks

### PR DESCRIPTION
**Pull request by flrsh**
[INPUT] [REWORK] [WIKI]

## About the PR 

This PR concerns the motives system on the RP servers. It adds debuffs in the form of status effects (which will be visible in the upper right like other status effects!) when your hunger or thirst meters drop below 25. The hunger debuff reduces your maximum health by 20, while the thirst debuff reduces your stamina regeneration by 3. Once your meter goes above 25, the debuffs are removed. This PR also removes the dropped click mechanic because I think it's really unintuitive and unfun gameplay, and also because I think these debuffs should be enough. 

I'd love players' input on these numbers and whether they think they'll be okay. An idea I had was that once you receive a debuff at 25, you have to eat or drink enough to get your motive back up to 50, but I'm not sure how different that would be in terms of actual gameplay. 

## Why's this needed? 

I think this is definitely more engaging than dropped clicks, and also adds a bit of risk in not eating or drinking enough. 

## Changelog

```
(u)Flourish:
(*)The motives system on RP has been slightly reworked. Instead of dropping clicks when your overall motives are too low, you now receive a maximum health debuff when your hunger is too low, and a stamina regeneration debuff when your thirst is too low. 
```

### CODE REVIEW

**Comment by Yellow-Mushroom** [comment]
This is needed so bad, tbh I’d support even harsher debuffs because I believe if you aren’t taking breaks to eat and drink you aren’t having as much fun as you could, and if you’re too busy to eat or drink, then it should very much show in your ability. 

Also dropped clicks are hell, especially for lagging players, they are the antithesis of fun or balance. 

Tl;dr: this needs to be added ASAP, and further balancing and come after this is shoved in

**Review by ZeWaka — APPROVED** [review]

File: code/datums/sims.dm [hunk 1]
```diff
@@ -160,17 +160,21 @@
 		icon_state = "hunger"
 		desc = "Hunger can be raised by eating various edible items, more complex dishes raise your hunger more."
 		depletion_rate = 0.078
-		var/starve_message = "<span class='alert'>You are starving to death!</span>"
+		var/debuff = "hungry"
+		var/debuff_threshold = 25
+		var/debuffed = 0
```
Comment (ZeWaka) [inline code/datums/sims.dm:165]:
```suggestion
		var/debuffed = FALSE
```

File: code/datums/sims.dm [hunk 2]
```diff
@@ -160,17 +160,21 @@
...
-
-		onDeplete()
-			if (!starving)
-				showOwner(starve_message)
-				starving++
+		onIncrease()
+			if (value > debuff_threshold && debuffed)
+				showOwner("<span class='notice'>You feel considerably less [debuff]!</span>")
+				holder.owner.delStatus(debuff)
+				debuffed = 0
```
Comment (ZeWaka) [inline code/datums/sims.dm:171]:
```suggestion
				debuffed = FALSE
```

File: code/datums/sims.dm [hunk 3]
```diff
@@ -160,17 +160,21 @@
...
 
-		onFill()
-			starving = 0
+		onDecrease()
+			if (value < debuff_threshold && !debuffed)
+				showOwner("<span class='alert'>You feel considerably [debuff]!</span>")
+				holder.owner.setStatus(debuff, duration = null)
+				debuffed = 1
```
Comment (ZeWaka) [inline code/datums/sims.dm:177]:
```suggestion
				debuffed = TRUE
```

**Comment by Emcee-Gore** [comment]
I enjoyed the harsher click dropping debuffs as much as I spited them.
Most of the time they were the only reason to remind me of taking a break in the bar.

I fear that some health debuff and just the lower stamina are not bad enough to get someone to charge up their body through mortal food.

The additional transperency concerning debuffs is great.
I'd perhaps enjoy some tweaking to the click dropping debuff, but no straight up removal.

**Comment by flrsh** [comment]
Since pretty much everyone is in favor of this change, I'm gonna merge it now instead of waiting a couple more days!

> This is needed so bad, tbh I’d support even harsher debuffs because I believe if you aren’t taking breaks to eat and drink you aren’t having as much fun as you could, and if you’re too busy to eat or drink, then it should very much show in your ability.

>I fear that some health debuff and just the lower stamina are not bad enough to get someone to charge up their body through mortal food.

I changed the thirst debuff to reduce stamina regen by 5, since people seemed generally okay with that idea! This should hopefully compel people to hydrate more. 

> I'd perhaps enjoy some tweaking to the click dropping debuff, but no straight up removal.

I think the majority of people want it removed, so I probably won't be adding them back.
